src/ui/CategoryFacet/CategoryFacet.ts

```diff
@@ -616,6 +648,45 @@ export class CategoryFacet extends Component implements IAutoLayoutAdjustableIns
     $$(this.element).removeClass('coveo-hidden');
   }
 
+  /**
+   * Collapses or expands the facet depending on it's current state.
+   */
+  public toggleCollapse() {
+    this.isCollapsed ? this.expand() : this.collapse();
+  }
+
+  /**
+   * Expands the facet, displaying all of its currently fetched values.
+   */
+  public expand() {
+    if (!this.options.enableCollapse) {
+      return this.logger.warn(`Calling expand() won't do anything on a facet that has the option "enableCollapse" set to "false"`);
+    }
+    if (!this.isCollapsed) {
```

Comment:
Debatable, but I think it would be better to execute the code even if the internal state variable `this.isCollapsed` says the facet is already expanded.

My reasoning is that let's say there is a bug around that state variable and it gets out of sync with the facet actual appearance to the end user.

The facet would get "stuck" in that mode.